astropy/wcs/tests/test_utils.py

```diff
@@ -561,13 +568,17 @@ def test_noncelestial_scale(cdelt, pc, cd):
         mywcs.wcs.cd = cd
     if pc is not None:
         mywcs.wcs.pc = pc
-    mywcs.wcs.cdelt = cdelt
 
-    mywcs.wcs.ctype = ['RA---TAN', 'FREQ']
+    # Some inputs emit RuntimeWarning
```

Comment:
The warnings here are because the parameters are set incorrectly. 
A WCS cannot have both `CD` and `PC` defined.  A WCS cannot have `CD` and `CDELT` defined.
Since the test is not meant to test the warning messages I think the input data is incorrect.
Either fix the inputs to the test and remove warnings or file an issue and I will fix it in a different PR.
I remember seeing some function code that needs to be fixed in a similar way but I have to find it.

Reply:
@nden , I removed the handling of `RuntimeWarning`. Can we proceed and merge this soon?

Reply:
I also opened an issue at #9310 as you requested. Thanks!